src/pyqasm/scope.py

```
@@ -149,19 +149,19 @@ def check_in_scope(self, var_name: str) -> bool:
-            or self.in_box_scope()
-            or self.in_pulse_scope()
-        ):
+        if self.in_function_scope() or self.in_gate_scope() or self.in_pulse_scope():
             if var_name in curr_scope:
                 return True
             if var_name in global_scope:
                 return global_scope[var_name].is_constant or global_scope[var_name].is_qubit
-        if self.in_block_scope():
+        if self.in_block_scope() or self.in_box_scope():
+            in_box = self.in_box_scope()
             for scope, context in zip(reversed(self._scope), reversed(self._context)):
```

Ruff (0.15.7)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)